modules/oe_theme_content_event/src/Plugin/ExtraField/Display/StatusMessage.php

```diff
@@ -0,0 +1,202 @@
+
+/**
+ * Extra field displaying the event status messages.
+ *
+ * @ExtraFieldDisplay(
+ *   id = "oe_theme_content_event_status_message",
+ *   label = @Translation("Status message"),
+ *   bundles = {
+ *     "node.oe_event",
+ *   },
+ *   visible = true
+ * )
+ */
+class StatusMessage extends DateAwareExtraFieldBase {
+
+  /**
+   * Date formatter service.
+   *
+   * @var \Drupal\Core\Datetime\DateFormatterInterface
+   */
+  protected $dateFormatter;
+
+  /**
+   * ProgrammeExtraField constructor.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   * @param \Drupal\Component\Datetime\TimeInterface $time
+   *   Time service.
+   * @param \Drupal\oe_time_caching\Cache\TimeBasedCacheTagGeneratorInterface $cache_tag_generator
+   *   Time based cache tag generator service.
+   * @param \Drupal\Core\Datetime\DateFormatterInterface $date_formatter
+   *   The date formatter.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, EntityTypeManagerInterface $entity_type_manager, TimeInterface $time, TimeBasedCacheTagGeneratorInterface $cache_tag_generator, DateFormatterInterface $date_formatter) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition, $entity_type_manager, $time, $cache_tag_generator);
+    $this->dateFormatter = $date_formatter;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('entity_type.manager'),
+      $container->get('datetime.time'),
+      $container->get('oe_time_caching.time_based_cache_tag_generator'),
+      $container->get('date.formatter')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getLabel() {
+    return $this->t('Status message');
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+   * @SuppressWarnings(PHPMD.NPathComplexity)
+   */
+  public function viewElements(ContentEntityInterface $entity) {
+    // Render event status messages using the ECL Message component.
+    $event = new EventNodeWrapper($entity);
+    $build = ['#theme' => 'oe_theme_content_event_status_message'];
+
+    $cacheable = CacheableMetadata::createFromRenderArray($build);
+
+    // If the event is not over then apply time-based tags, so that it can be
+    // correctly invalidated once the event is over.
+    if (!$event->isOver($this->requestDateTime)) {
+      $this->applyHourTag($build, $event->getEndDate());
```

Comment:
Logic issue, I set the event as planned with the event dates matching the online dates with event cancelled, the message is correct, but when I set it to Rescheduled, the event shows the message "This event has started. You can also watch it via livestream." even when I set the event state to cancelled or postponed, the message doesn't change, even when I'm logged in as admin or logged out. This is due to the fact that you don't exit here, and it continues until the line 147. Please make sure such case is covered by test after fixing it.

Reply:
We'll need confirmation from PO.

Reply:
Updated the logic.